app/controllers/shared/api.php

```diff
@@ -49,44 +50,164 @@
+                    $buildsStorage = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".builds.storage"));
+                    $buildsCompute = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".builds.compute"));
+                    $executions = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".executions"));
+                    $executionsCompute = $dbForProject->getDocument('stats', md5("_inf_" . "{$document->getId()}" . ".executions.compute"));
+
+                    $queueForUsage
+                        ->addMetric("deployments", ($deployments['value'] * -1))
+                        ->addMetric("deployments.storage", ($deploymentsStorage['value'] * -1))
+                        ->addMetric("builds", ($builds['value'] * -1))
+                        ->addMetric("builds.storage", ($buildsStorage['value'] * -1))
+                        ->addMetric("builds.compute", ($buildsCompute['value'] * -1))
+                        ->addMetric("executions", ($executions['value'] * -1))
+                        ->addMetric("executions.compute", ($executionsCompute['value'] * -1))
+                    ;
+                }
+                break;
+            case $document->getCollection() === 'deployments':
+                $queueForUsage
+                    ->addMetric("deployments", $value) // per project
+                    ->addMetric("deployments.storage", $document->getAttribute('size') * $value) // per project
+                    ->addMetric("{$document['resourceType']}" . "." . "{$document['resourceId']}" . ".deployments", $value)// per function
+                    ->addMetric("{$document['resourceType']}" . "." . "{$document['resourceId']}" . ".deployments.storage", $document->getAttribute('size') * $value) // per function
+                ;
+                break;
+            case $document->getCollection() === 'executions':
+                $queueForUsage
+                    ->addMetric("executions", $value) // per project
+                    ->addMetric("{$document['functionId']}" . ".executions", $value) // per function
+                ;
+                break;
+            default:
+                break;
+        }
+    } catch (Throwable $error) {
+        if (!empty($logger) && $error->getCode() === 500) {
+            $log = new Log();
+            $isProduction = App::getEnv('_APP_ENV', 'development') === 'production';
+            $log
+                ->setNamespace("appwrite-worker")
+                ->setServer(\gethostname())
+                ->setVersion(App::getEnv('_APP_VERSION', 'UNKNOWN'))
+                ->setType(Log::TYPE_ERROR)
+                ->setMessage($error->getMessage())
+                ->setAction('appwrite-queue-usage')
+                ->addTag('verboseType', get_class($error))
+                ->addTag('code', $error->getCode())
+                ->addExtra('collection', $document->getCollection())
```

Comment:
can we also make sure, we add event, so we know which event failed as well

Reply:
if you know the collection then you know what metric failed, am i wrong?